### PR TITLE
Image Banner: use container queries to support usage in kirby-grid and kirby-slides

### LINKED ISSUE #3663: [Enhancement] Extensions: Image-banner using container queries instead.
- [ ] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the enhancement
The image-banner as is is not suited for use in kirby-grid nor in kirby-slider, since the visual representation switches with media queries.

## Describe the solution you'd like

Make use of container queries instead so you can achieve the behaviour as represented in enclosed video.

## Are there any additional context?
https://github.com/user-attachments/assets/00500dcd-f3af-49c0-a5ae-ad7381d6579a

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Refinement

- [ ] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#ux-refinement)
- [ ] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [ ] Make sure you have read: "[Before you get coding](../CONTRIBUTING.md/#before-you-get-coding)".
- [ ] Signal to others you are working on the issue by assigning yourself.
- [ ] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [ ] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [ ] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".

### PR DESCRIPTION
**Pull request by troelslenda**
## Which issue does this PR close?

This PR closes #3663 

## What is the new behavior?
Use container queries for responsive behavior instead of relying on viewport width. This will support usage in kirby-grid, kirby-slides and many other usecases. 

Displayed in the new storybook story.

## Does this PR introduce a breaking change?

- [x] Yes
- [ ] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Reminders
- [ ] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [ ] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [ ] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [ ] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [ ] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: libs/extensions/angular/image-banner/src/image-banner.component.scss [hunk 1]
```diff
@@ -5,6 +5,25 @@ $blur-radius: 60px;
 $max-content-width: 324px;
 $main-image-height-small: 132px;
 $main-image-height-large: 164px;
+$cq-width: 515px;
+
+@mixin cq-min() {
```
Comment (RasmusKjeldgaard) [inline libs/extensions/angular/image-banner/src/image-banner.component.scss]: consider a more descriptive name, it loses a bit of meaning like this.
Maybe `container-below-min-width` and `container-above-min-width`

File: libs/extensions/angular/image-banner/src/image-banner.component.stories.ts [hunk 2]
```diff
@@ -106,3 +106,43 @@ export const NoDismiss: Story = {
     template: `<kirby-x-image-banner ${argsToTemplate(args)}></kirby-x-image-banner>`,
   }),
 };
+
+export const containerQueries: Story = {
```
Comment (RasmusKjeldgaard) [inline libs/extensions/angular/image-banner/src/image-banner.component.stories.ts]: Please add jsdoc similar to other stories, explaining what the user sees in the example. We use this for documentation in storybook.

File: libs/extensions/angular/image-banner/src/image-banner.component.stories.ts [hunk 3]
```diff
@@ -106,3 +106,43 @@ export const NoDismiss: Story = {
...
+  args: {
+    title: 'A title of',
```
Comment (RasmusKjeldgaard) [inline libs/extensions/angular/image-banner/src/image-banner.component.stories.ts]: Please add a descriptive title to the banner

File: libs/extensions/angular/image-banner/src/image-banner.component.stories.ts [hunk 4]
```diff
@@ -106,3 +106,43 @@ export const NoDismiss: Story = {
...
+        </div>
+        <div class="kirby-grid-item" xs="3">
+          <kirby-x-image-banner ${argsToTemplate(args)}></kirby-x-image-banner>
+        </div>
+        <div class="kirby-grid-item" xs="3">
+          <kirby-x-image-banner ${argsToTemplate(args)}></kirby-x-image-banner>
+        </div>
+        <div class="kirby-grid-item" xs="3">
+          <kirby-x-image-banner ${argsToTemplate(args)}></kirby-x-image-banner>
+        </div>
```
Comment (RasmusKjeldgaard) [inline libs/extensions/angular/image-banner/src/image-banner.component.stories.ts]: Does this last row provide any value in docs or screenshot testing?
It does not showcase too well on mobile, consider just having two rows as I think it still shows your intention.

File: libs/extensions/angular/image-banner/src/image-banner.component.stories.ts [hunk 5]
```diff
@@ -106,3 +106,43 @@ export const NoDismiss: Story = {
...
+
+export const containerQueries: Story = {
+  args: {
+    title: 'A title of',
+    bodyText: 'Some body text.',
+    actionButtonText: 'Read more',
+    imagePath: 'assets/images/leaves.jpg',
+    dismissClick: undefined,
+  },
+  // The render method with argsToTemplate() is needed for bannerDismiss to not be automatically inferred by storybook.
```
Comment (RasmusKjeldgaard) [inline libs/extensions/angular/image-banner/src/image-banner.component.stories.ts]: Consider removing this comment, it does not really make sense for this snippet.

File: libs/extensions/angular/image-banner/src/image-banner.component.scss [hunk 6]
```diff
@@ -5,6 +5,25 @@ $blur-radius: 60px;
 $max-content-width: 324px;
 $main-image-height-small: 132px;
 $main-image-height-large: 164px;
+$cq-width: 515px;
```
Comment (RasmusKjeldgaard) [inline libs/extensions/angular/image-banner/src/image-banner.component.scss]: Is this a semi-arbitrary value, or is it calculated based on something?

File: libs/extensions/angular/image-banner/src/image-banner.component.stories.ts (hunk 4 above)
Comment (RasmusKjeldgaard) [reply libs/extensions/angular/image-banner/src/image-banner.component.stories.ts]: image

File: libs/extensions/angular/image-banner/src/image-banner.component.stories.ts (hunk 2 above)
Comment (RasmusKjeldgaard) [reply libs/extensions/angular/image-banner/src/image-banner.component.stories.ts]: Something about how the component adapts to the containers width, and thus should be plug and play with the kirby css grid utilities.

Comment (RasmusKjeldgaard) [inline libs/extensions/angular/image-banner/src/image-banner.component.stories.ts]:
```suggestion
export const UsageInGrid: Story = {
```

File: libs/extensions/angular/image-banner/src/image-banner.component.scss (hunk 1 above)
Comment (RasmusKjeldgaard) [inline libs/extensions/angular/image-banner/src/image-banner.component.scss]: Perfectly fine to leave the container queries in here to start with, then we can promote them to the design system at a later point 👍

**Review by RasmusKjeldgaard — COMMENTED** [review]
Great work, I have some comments you can consider. 
Most of it are just suggestions, but I would prefer if the docs-related comments are taken care of before I approve.

File: libs/extensions/angular/image-banner/src/image-banner.component.scss (hunk 6 above)
Comment (troelslenda) [reply libs/extensions/angular/image-banner/src/image-banner.component.scss]: The value was read from the width of `ImageBannerComponent` in the default story where the media query small   triggered.

Comment (RasmusKjeldgaard) [reply libs/extensions/angular/image-banner/src/image-banner.component.scss]: I think this value should be 600px 🤔 
When I inspect the original implementation in the [default example](https://extensions.kirby.design/?path=/story/components-banner-image-banner--default), it changes when the iframe is 632px wide. And this setup mimics spacing in `kirby-page` with 16px padding left and right, so the container around the banner is 600px when the layout change happens.

Comment (troelslenda) [reply libs/extensions/angular/image-banner/src/image-banner.component.scss]: done :)

File: libs/extensions/angular/image-banner/src/image-banner.component.stories.ts [hunk 7]
```diff
@@ -106,3 +106,35 @@ export const NoDismiss: Story = {
...
+ * play with the kirby css grid and slider utilities.
+ * It switches between a narrow and wide view.
+ */
+export const UsageInGrid: Story = {
+  args: {
+    title: 'Banners that adapt in grid',
+    bodyText: 'This is the body text.',
+    actionButtonText: 'Read more',
+    imagePath: 'assets/images/leaves.jpg',
+    dismissClick: undefined,
```
Comment (RasmusKjeldgaard) [inline libs/extensions/angular/image-banner/src/image-banner.component.stories.ts]: @henrikvoetmand suggested to remove this so the dismiss button shows, and also get the externalLink button shown to get that use-case validated when we screenshot test.

```suggestion
    externalLink: 'http://www.kirby.design',
```

**Review by RasmusKjeldgaard — APPROVED** [review]
Thanks a lot for contributing this!

**Pull request merged**